cmd/minikube/cmd/root_test.go

```diff
@@ -44,3 +46,43 @@ func TestPreRunDirectories(t *testing.T) {
 		}
 	}
 }
+
+func cleanup() {
+	pflag.Set("v", "0")
+	pflag.Lookup("v").Changed = false
+}
+
+func TestFlagShouldOverrideConfig(t *testing.T) {
+	defer cleanup()
+	viper.Set("v", "1337")
+	pflag.Set("v", "100")
+	setFlagsUsingViper()
+	if viper.GetInt("v") != 100 {
+		viper.Debug()
+		t.Fatal("Value from viper config overrode explicit flag value")
+	}
+}
+
+func TestConfigShouldOverrideDefault(t *testing.T) {
+	defer cleanup()
+	viper.Set("v", "1337")
+	setFlagsUsingViper()
+	if viper.GetInt("v") != 1337 {
+		viper.Debug()
+		t.Fatalf("Value from viper config did not override default flag value")
+	}
+}
+
+func TestFallbackToDefaultFlag(t *testing.T) {
```

Comment:
Also try to switch these to table-driven: https://github.com/golang/go/wiki/TableDrivenTests#example-of-a-table-driven-test